core/include/crypto/crypto.h

```diff
@@ -22,6 +22,7 @@
 #ifndef __CRYPTO_CRYPTO_H
 #define __CRYPTO_CRYPTO_H
 
+#include <crypto/secure_element.h>
```

Comment:
I don't think this is needed here. Nothing in this file is supposed to depend on anything from `crypto/secure_element.h`.

Reply:
right, I didnt add it for dependency but for consistency as I think it should be part of the crypto api. 
For example, retrieving the rng from a secure element would be just sending an APDU frame request using this interface...should the user include the superset crypto.h instead of the secure_element.h?

Reply:
jfyi crypto_do_apdu (in the case of the se05x secure element) will - when SCP03 has been enabled- encrypt the raw APDU frame using the SPC03 keys using OP-TEE tomcrypt and send the frame to the linux kenel using the i2c trampoline

Reply:
if SCP03 was not enabled, it will just use the trampoline to send the request and receive the response.

Reply:
To me it's as much part of the crypto api as `crypto/crypto_accel.h` is (which isn't included here).
There's a value in not exposing more than needed. I know this is a tradeoff, in this case I believe it makes sense to draw the line here.

Reply:
right, and that makes little sense to me as well although is kind of different as the prefix does some of the changes ...

but anyway, ok, so to follow that same pattern I will have to rename the file to 
- crypto_secure_element.h

and the interfaces:
- crypto_secure_element_enable_scp03 
- crypto_secure_element_do_apdu

ok?

Reply:
Well, if you insist. I would have kept the function names as they are, I just wanted to get it out of the way of the "common case".